optuna/pruners/successive_halving.py

```diff
@@ -103,9 +119,17 @@ def prune(self, study, trial):
 
         rung = _get_current_rung(trial)
         value = trial.intermediate_values[step]
-        trials = None
+        trials = None  # type: Optional[List[FrozenTrial]]
 
         while True:
+            if self._min_resource is None:
+                if trials is None:
+                    trials = study.get_trials(deepcopy=False)
+                self._estimate_min_resource(trials)
```

Comment:
How about making `_estimate_min_resource` a free function that return the value insetting of setting it to `self`? I think it improved readability.

Reply:
Ah, you're right. This conceals that `self._estimate_min_resource` modifies `self._min_resource`.